import java.util.Scanner;

public class ExceptionsDemo {
    public static void main(String[] args) {

        // Example 1
        // Arithmetic Exception because input is 0.

//        System.out.println("program is started...");
//          Scanner sc = new Scanner(System.in);
//        System.out.println("Enter a number: ");
//        int input1 = sc.nextInt();
//
//        System.out.println(100/input1);
//
//
//        System.out.println("Program is completed...");
//        System.out.println("Program is exited...");


        // Example 2

        // Input = 5, so it wil throw an ArrayOutOfBounds Exception

//        int a[] = new int[5];
//
//        System.out.println("Enter a position(0-4):");
//        int position = sc.nextInt();
//
//        System.out.println("Enter a Value : ");
//        int value = sc.nextInt();
//
//        a[position] = value; // this line will throw an exception
//        System.out.println(a[position]);
//
//        System.out.println("Program is completed...");
//        System.out.println("Program is exited...");

        // Example 3

        // Instead of number, if we put alphabet then will it convert into num? Noo

//        String s = "12345";

//        String s = "Welcome"; // Number Format Exception

        // Convert from number to String format
//        int num = Integer.parseInt(s);
//        System.out.println(num);

//        System.out.println("Program is completed...");
//        System.out.println("Program is exited...");


        // Example 4  Null Pointer Exception

//        String s1 = "welcome"; // or null
//        System.out.println(s1.length());

//        String s1 = "";
//        System.out.println(s1.length());

//        String s1 = null;
//        System.out.println(s1.length());

    }
}
